labs/lab08/sol/lin/3-safety/malloc.c

```diff
@@ -0,0 +1,156 @@
+/**
+ * SO
+ * Lab #8
+ *
+ * Task #3, lin
+ *
+ * Is malloc thread safe?
+ */
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <pthread.h>
+#include <unistd.h>
+
+#include <utils.h>
+#include "critical.h"
+
+#define NUM_THREADS 10
+#define NUM_ROUNDS  10000
+#define NUM_ROWS    100
+#define NUM_COLUMNS 1000
+
+#define PRINT_HDR_SZ    32
+#define PRINT_BUF_SZ    256
+#define MAIN_PRINT      (-1)
+
+int global_storage;
+
+static void increase_numbers(int *a, int *b, int*c) {
+	(*a)++;
+	(*b)++;
+	(*c)++;
+}
+
+static void print_stats(int origin,
+						int *global_storage,
+						int *function_global_storage,
+						int *function_specific_storage)
```

Comment:
```suggestion
			int *global_storage,
			int *function_global_storage,
			int *function_specific_storage)
```

It just looked weird like that.